### ISSUE TITLE
aria-roles rule does not support space separated tokens

### ISSUE THREAD

**Issue by besenwagen** [dequelabs/axe-core#3606]
### Product

axe-core

### Product Version

4.4.3

### Lastest Version

- [X] I have tested the issue with the latest version of the product

### Issue Description

In the a11y slack somebody recently asked wether Axe failing space separated token values for the `role` attribute is a WCAG or a best practices failure. I assume they used a GUI that doesn't disclose details, axe-core fails it under WCAG SC 4.1.2.

The axe-core **aria-roles** rule
https://github.com/dequelabs/axe-core/blob/develop/doc/rule-descriptions.md#wcag-20-level-a--aa-rules

should implement

ACT rule **Role attribute has valid value**
https://act-rules.github.io/rules/674b10

which clearly describes space separated tokens and has a passing test example for it. 

#### Expectation

The ACT rule's passing test example should not cause a WCAG error in axe-core.

#### Actual

The ACT rule's passing test example causes a WCAG error in axe-core.

#### How to Reproduce

Create a test page that includes the ACT example for a passing test case with space separated tokens in the `role` attribute value and use it as input for axe-core.

**Comment by besenwagen** [comment]
> Auto assign axe-core integration issues to Axe API team project board: All jobs have failed

**Comment by WilcoFiers** [comment]
Axe-core fails this because of issues with Internet Explorer. We're planning to change this fairly soon, since IE is now no longer officially supported.

**Comment by besenwagen** [comment]
Thanks. I don't really understand that, though. It’s categorized as a WCAG SC failure, shouldn’t issues related to browser support be put under best practices? Maybe I have some basic misunderstanding there.

**Comment by aardrian** [comment]
I think this change should also address the _aria-allowed-attr_ rule and maybe the _region_ rule, Right now, the space-separated roles seem to confuse the check for what attributes are allowed.

For example, this passes an axe test:
```
<div aria-label="test" role="dialog">
  Test
</div>
```

As does this:
```
<div aria-label="test" role="region">
  Test
</div>
```

But when I add the space-separated roles as such:
```
<div aria-label="test" role="dialog region">
  Test
</div>
```

It fails:
- [aria-roles](https://dequeuniversity.com/rules/axe/4.4/aria-roles)
- [aria-allowed-attr](https://dequeuniversity.com/rules/axe/4.4/aria-allowed-attr)
- [region](https://dequeuniversity.com/rules/axe/4.4/region) (which I suspect is related)

Try other roles, like `group` and `region`, or whatever. I made a pen so you can see it in action: https://codepen.io/aardrian/pen/ZExVQxp

**Comment by WilcoFiers** [comment]
Going to create a split here. For axe-core 4.5 we're going to start allowing fallback roles. Full support for them will have to wait until axe-core 4.6 though, as it requires a number of fairly fundamental changes to how axe selects nodes.

**Comment by padmavemulapati** [comment]
Validated with the latest axe-core develop branch code base, 
with the testscript:
failing on
```
<div role="doc-biblioentry" id="fail1">fail</div>
  <div role="doc-endnote" id="fail2">fail</div>
```
but passing when role="presentation non" (passing role with space seperated text) - as fallbackrole is no longer using
```
<div role="presentation none" id="pass2"></div>
  <div role="button alert" id="pass1">fail</div>
```


![Image](https://user-images.githubusercontent.com/47409579/194854719-fab54d54-dac1-4806-afe6-19c47527f88d.png)
